### DISCUSSION TITLE
How to Automatically Start Cronicle on Mac Boot

### DISCUSSION THREAD

**Discussion by mfranzel** [jhuckaby/Cronicle#835] [Q&A]
Hello,
I was hoping someone could help.

I installed Cronicle on my Mac mini M4 and it is working great! It's an amazing tool! However, I was wondering if there is a way to make it start automatically when the Mac turns on?

**Comment by jhuckaby** [comment]
Does this still work?

https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#server-startup

It used to.

I haven't tried it on macOS Sonoma or Sequoia yet, and certainly not on an M4, so it's possible Apple is now thwarting the old LaunchAgent plist trick, which this uses.

**Reply by mfranzel** [reply]
Nope... didn't work :(

**Reply by jhuckaby** [reply]
Ah crap, Apple must have changed something.  I'll look into it as soon as I have some time.

Here's the code in question -- it used to work:

https://github.com/jhuckaby/pixl-boot/blob/master/boot.js#L207-L245

**Comment by jhuckaby** [comment]
This should be fixed in v0.9.62.

https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.62

Upgrade instructions are here: https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#upgrading-cronicle

Remember that on macOS the boot behavior will be different depending on if you run the `npm run boot` command as root (recommended), or as your standard macOS user:

- If you run `npm run boot` as root or sudo, which is the recommended way, then Cronicle will be installed as a system-level LaunchDaemon, and will start when the Mac first boots up, before any user logs in.  It will run as root, which is how it was designed to run.
- If you run `npm run boot` as your user account, then Cronicle will be installed as a local LaunchAgent, and only start when you actually login, and the macOS Finder / Desktop loads up.  It will run as your user, and not root.  Cronicle will also not be able to switch users for jobs in this mode.

**Reply by mfranzel** [reply]
I updated per your instructions, which was successful. I changed directories to`/opt/cronicle` while as root. I ran `npm run boot` and the output said 

```

> Cronicle@0.9.62 boot
> pixl-boot install

Installing startup service: Cronicle...OK.
Successfully registered startup service.

```

I rebooted but Cronicle did not start up automatically. I tried it as my user as well, but still no dice.

**Reply by jhuckaby** [reply]
Hmmm..... this works perfectly on my Mac Mini M2 with macOS Sonoma.  I wonder what's going on with your situation.  I guess it could be something weird with macOS Sequoia, but OH....  did you install Node.js using the macOS installer (GUI app), so your Node binary is located here:

```
/usr/local/bin/node
```

It needs to exist there, or at `/usr/bin/node`, for the auto-boot startup thing to work.

It will *not* work with any of those "Node.js version managers" (nvm and family) because those install Node.js to a weird location.

If all that is correct on your Mac, then we need to dig deeper and look at the logs.  Try clearing out all the logs in `/opt/cronicle/logs/*.log` and then doing a reboot.  See if any new logs appear in that directory at startup.  Perhaps it's *trying* to start but encountering some kind of error?
